Index: FFmpeg/libavcodec/decode.c
===================================================================
--- FFmpeg.orig/libavcodec/decode.c
+++ FFmpeg/libavcodec/decode.c
@@ -2329,6 +2329,8 @@ void ff_decode_internal_sync(AVCodecCont
     const DecodeContext *src_dc = decode_ctx(src->internal);
     DecodeContext *dst_dc = decode_ctx(dst->internal);
 
+    dst_dc->initial_pict_type = src_dc->initial_pict_type;
+    dst_dc->intra_only_flag   = src_dc->intra_only_flag;
     ff_refstruct_replace(&dst_dc->lcevc, src_dc->lcevc);
 }
 
Index: FFmpeg/libavcodec/pthread_frame.c
===================================================================
--- FFmpeg.orig/libavcodec/pthread_frame.c
+++ FFmpeg/libavcodec/pthread_frame.c
@@ -115,10 +115,6 @@ typedef struct PerThreadContext {
     int hwaccel_threadsafe;
 
     atomic_int debug_threads;       ///< Set if the FF_DEBUG_THREADS option is set.
-
-    /// The following two fields have the same semantics as the DecodeContext field
-    int intra_only_flag;
-    enum AVPictureType initial_pict_type;
 } PerThreadContext;
 
 /**
@@ -821,13 +817,6 @@ static av_cold int init_thread(PerThread
     AVCodecContext *copy;
     int err;
 
-    p->initial_pict_type = AV_PICTURE_TYPE_NONE;
-    if (avctx->codec_descriptor->props & AV_CODEC_PROP_INTRA_ONLY) {
-        p->intra_only_flag = AV_FRAME_FLAG_KEY;
-        if (avctx->codec_type == AVMEDIA_TYPE_VIDEO)
-            p->initial_pict_type = AV_PICTURE_TYPE_I;
-    }
-
     atomic_init(&p->state, STATE_INPUT_READY);
 
     copy = av_memdup(avctx, sizeof(*avctx));
