Index: FFmpeg/libavformat/sdp.c
===================================================================
--- FFmpeg.orig/libavformat/sdp.c
+++ FFmpeg/libavformat/sdp.c
@@ -322,7 +322,7 @@ static int extradata2psets_hevc(AVFormat
             pos += len;
         }
     }
-    if (!ps_pos[0] || !ps_pos[1] || !ps_pos[2])
+    if (!ps_pos[1] || !ps_pos[2])
         goto err;
 
     psets = av_mallocz(MAX_PSET_SIZE);
@@ -335,6 +335,8 @@ static int extradata2psets_hevc(AVFormat
 
     for (i = 0; i < 3; i++) {
         pos = ps_pos[i];
+        if (!pos)
+            continue;
 
         if (i > 0)
             av_strlcat(psets, "; ", MAX_PSET_SIZE);
