diff --git a/libavfilter/qsvvpp.c b/libavfilter/qsvvpp.c
index 0818ada1..e02c8274 100644
--- a/libavfilter/qsvvpp.c
+++ b/libavfilter/qsvvpp.c
@@ -467,6 +467,17 @@ static QSVFrame *submit_frame(QSVVPPContext *s, AVFilterLink *inlink, AVFrame *p
     else if (qsv_frame->frame->repeat_pict == 4)
         qsv_frame->surface.Info.PicStruct |= MFX_PICSTRUCT_FRAME_TRIPLING;
 
+    // Apply crop using AVFrame crop fields
+    if (qsv_frame->surface.Info.CropX == 0 && qsv_frame->surface.Info.CropY == 0 &&
+        qsv_frame->surface.Info.CropW == picref->width && qsv_frame->surface.Info.CropH == picref->height) {
+        if (picref->crop_left > 0 || picref->crop_top > 0 || picref->crop_right > 0 || picref->crop_bottom > 0) {
+            qsv_frame->surface.Info.CropW = (mfxU16)((picref->width - picref->crop_right) - picref->crop_left);
+            qsv_frame->surface.Info.CropH = (mfxU16)((picref->height - picref->crop_bottom) - picref->crop_top);
+            qsv_frame->surface.Info.CropX = (mfxU16)picref->crop_left;
+            qsv_frame->surface.Info.CropY = (mfxU16)picref->crop_top;
+        }
+    }
+
     return qsv_frame;
 }
 
